src/Featurizers/AnalyticalRollingWindowFeaturizer.h

```diff
@@ -0,0 +1,337 @@
+        [&ar](void) {
+            // Version
+            std::uint16_t                   majorVersion(Traits<std::uint16_t>::deserialize(ar));
+            std::uint16_t                   minorVersion(Traits<std::uint16_t>::deserialize(ar));
+
+            if(majorVersion != 1 || minorVersion != 0)
+                throw std::runtime_error("Unsupported archive version");
+
+            // Because we know that maxWindowSize and minWindowSize come in originally as uint32, this cast is always safe
+            std::uint32_t                       maxWindowSize(Traits<std::uint32_t>::deserialize(ar));
+            AnalyticalRollingWindowCalculation  windowCalculation(static_cast<AnalyticalRollingWindowCalculation>(Traits<std::uint8_t>::deserialize(ar)));
+            std::uint32_t                       horizon(Traits<std::uint32_t>::deserialize(ar));
+            std::uint32_t                       minWindowSize(Traits<std::uint32_t>::deserialize(ar));
+
+
+            return AnalyticalRollingWindowTransformer(std::move(maxWindowSize), std::move(windowCalculation), std::move(horizon), std::move(minWindowSize));
+        }()
+    ) {
+}
+
+template <typename InputT, size_t MaxNumTrainingItemsV>
+bool AnalyticalRollingWindowTransformer<InputT, MaxNumTrainingItemsV>::operator==(AnalyticalRollingWindowTransformer const &other) const {
+    return this->_maxWindowSize  == other._maxWindowSize && this->_windowCalculation == other._windowCalculation && this->_horizon == other._horizon && this->_minWindowSize == other._minWindowSize;
+}
+
+template <typename InputT, size_t MaxNumTrainingItemsV>
+bool AnalyticalRollingWindowTransformer<InputT, MaxNumTrainingItemsV>::operator!=(AnalyticalRollingWindowTransformer const &other) const {
+    return (*this == other) == false;
+}
+
+template <typename InputT, size_t MaxNumTrainingItemsV>
+void AnalyticalRollingWindowTransformer<InputT, MaxNumTrainingItemsV>::save(Archive &ar) const /*override*/ {
+    // Version
+    Traits<std::uint16_t>::serialize(ar, 1); // Major
+    Traits<std::uint16_t>::serialize(ar, 0); // Minor
+
+    // Data
+    Traits<std::uint32_t>::serialize(ar, this->_maxWindowSize);
+    Traits<std::uint8_t>::serialize(ar, static_cast<std::uint8_t>(_windowCalculation));
+    Traits<std::uint32_t>::serialize(ar, this->_horizon);
+    Traits<std::uint32_t>::serialize(ar, this->_minWindowSize);
+
+    // Note that we aren't serializing working state
+}
+
+// ----------------------------------------------------------------------
+// |
+// |  AnalyticalRollingWindowEstimator
+// |
+// ----------------------------------------------------------------------
+template <typename InputT, size_t MaxNumTrainingItemsV>
+AnalyticalRollingWindowEstimator<InputT, MaxNumTrainingItemsV>::AnalyticalRollingWindowEstimator(AnnotationMapsPtr pAllColumnAnnotations, std::uint32_t maxWindowSize, AnalyticalRollingWindowCalculation windowCalculation, std::uint32_t horizon, std::uint32_t minWindowSize) :

```

Comment:
With this change, it would probably make sense to have the `maxWindowSize` and `minWindowSize` parameters appear next to each other. Also the members should be next to each other as well.

Reply:
The problem is that the `minWindowSize` parameter has a default value (in the declaration), so it needs to be at the end. Would it be better to move `maxWindowSize` next to it at the end then?

Reply:
Moved them together. Resolving.